source/extensions/tracers/zipkin/zipkin_core_types.cc

```diff
@@ -219,6 +238,41 @@ const std::string Span::toJson() {
   return json_string;
 }
 
+const zipkin::proto3::Span Span::toProtoSpan() const {
+  zipkin::proto3::Span span;
+  span.set_trace_id(traceIdAsByteString());
+  span.set_parent_id(parentIdAsByteString());
+  span.set_id(idAsByteString());
+  span.set_name(name_);
+
+  if (timestamp_) {
+    span.set_timestamp(timestamp_.value());
+  }
+
+  if (duration_) {
+    span.set_duration(duration_.value());
+  }
+
+  for (const auto& annotation : annotations_) {
+    if (annotation.isSetEndpoint()) {
+      if (annotation.value() == ZipkinCoreConstants::get().CLIENT_SEND) {
+        span.set_kind(zipkin::proto3::Span::CLIENT);
+        span.mutable_local_endpoint()->MergeFrom(annotation.endpoint().toProtoEndpoint());
+      } else if (annotation.value() == ZipkinCoreConstants::get().SERVER_RECV) {
+        span.set_kind(zipkin::proto3::Span::SERVER);
```

Comment:
@adriancole @jcchavezs could you help to clarify the use of `span.kind` here? Thanks!

Reply:
https://github.com/apache/incubator-zipkin-api/blob/master/zipkin.proto#L69-L73

Reply:
specifically, in v1 you needed annotations "sr" "ss" to say both that this is a server span and to allocate timestamp and duration. in v2 we use KInd, timestamp and duration. It is often the case that v2 spans have no annotations because the only ones they had were redundant with kind, timestamp and duration.

Reply:
TL;DR, if v2 we don't need this annotation but we need kind.

Reply:
Need to review `startSpan()` and `finishSpan()` (i.e. `void Span::finish()`).